File: demo/src/main/java/kaufland/com/demo/CommentActivity.java
package kaufland.com.demo;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;

import kaufland.com.coachbasebinderapi.PersistenceException;
import kaufland.com.demo.entity.ProductEntity;
import kaufland.com.demo.entity.UserCommentWrapper;
import kaufland.com.demo.entity.UserCommentWrapper.Companion.*;

public class CommentActivity extends AppCompatActivity {

    private static final String TAG = CommentActivity.class.getName();
    private ArrayAdapter<String> mAdapter;


    public static Intent buildIntent(MainActivity activity, String id) {
        Intent intent = new Intent(activity, CommentActivity.class);
        intent.putExtra("id", id);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_comment);
        final List<UserCommentWrapper> data = getParentEntity().getComments();

        mAdapter = new ArrayAdapter<String>(this, R.layout.comment_item_view, R.id.txt_comment, map(data)) {
            @NonNull
            @Override
            public android.view.View getView(final int position, @Nullable android.view.View convertView, @NonNull ViewGroup parent) {
                android.view.View view = super.getView(position, convertView, parent);
                view.findViewById(R.id.btn_delete).setOnClickListener(new android.view.View.OnClickListener() {
                    @Override
                    public void onClick(android.view.View v) {
                        data.remove(position);
                        try {
                            getParentEntity().builder().setComments(data).exit().save();
                        } catch (PersistenceException e) {
                            Log.e(TAG, "failed to save Entity", e);
                        }
                        recreate();
                    }
                });
                return view;
            }
        };

        ListView listView = findViewById(R.id.list);
        listView.setAdapter(mAdapter);

        findViewById(R.id.btn_post).setOnClickListener(new android.view.View.OnClickListener() {
            @Override
            public void onClick(android.view.View v) {
                List<UserCommentWrapper> mComments = getParentEntity().getComments();
                mComments.add(UserCommentWrapper.create().builder().
                        setComment(((EditText) findViewById(R.id.edit_text)).getText().toString()).
                        setUser("you").exit());
                try {
                    getParentEntity().builder().setComments(mComments).exit().save();
                    ((EditText) findViewById(R.id.edit_text)).setText("");
                    recreate();
                } catch (PersistenceException e) {
                    Log.e(TAG, "failed to save", e);
                }
            }
        });
    }

    private List<String> map(List<UserCommentWrapper> userCommentEntities) {
        List<String> result = new ArrayList<>();

        for (UserCommentWrapper entity : userCommentEntities) {
            result.add(entity.getComment() + "\n[" + entity.getUser() + "(" + entity.getAge() + ")]");
        }

        return result;
    }

    private ProductEntity getParentEntity() {
        return ProductEntity.create(getIntent().getStringExtra("id"));
    }
}

File: demo/src/main/java/kaufland/com/demo/CommentActivity.kt
package kaufland.com.demo

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.ListView
import kaufland.com.coachbasebinderapi.PersistenceException
import kaufland.com.demo.entity.ProductEntity
import kaufland.com.demo.entity.ProductEntity.Companion.create
import kaufland.com.demo.entity.UserCommentWrapper
import java.util.*

class CommentActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_comment)
        val data = parentEntity.comments?.toMutableList()
        val adapter =  object :ArrayAdapter<String?>(this, R.layout.comment_item_view, R.id.txt_comment, map(data)) {
            override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
                val view = super.getView(position, convertView, parent)
                view.findViewById<View>(R.id.btn_delete).setOnClickListener { v: View? ->
                    data?.removeAt(position)
                    try {
                        val entity = parentEntity
                        entity.comments = data
                        entity.save()
                    } catch (e: PersistenceException) {
                        Log.e(TAG, "failed to save Entity", e)
                    }
                    recreate()
                }
                return view
            }
        }
        val listView = findViewById<ListView>(R.id.list)
        listView.adapter = adapter
        findViewById<View>(R.id.btn_post).setOnClickListener { v: View? ->
            val mComments = parentEntity.comments?.toMutableList()
            mComments?.add(UserCommentWrapper.create().builder().setComment((findViewById<View>(R.id.edit_text) as EditText).text.toString()).setUser("you").exit())
            try {
                val entity = parentEntity
                entity.comments = mComments
                entity.save()
                (findViewById<View>(R.id.edit_text) as EditText).setText("")
                recreate()
            } catch (e: PersistenceException) {
                Log.e(TAG, "failed to save", e)
            }
        }
    }

    private fun map(userCommentEntities: List<UserCommentWrapper>?): List<String> {
        val result: MutableList<String> = ArrayList()
        for (entity in userCommentEntities!!) {
            result.add("""
    ${entity.comment}
    [${entity.user}(${entity.age})]
    """.trimIndent())
        }
        return result
    }

    private val parentEntity: ProductEntity
        private get() = create(intent.getStringExtra("id"))

    companion object {
        private val TAG = CommentActivity::class.java.name
        @JvmStatic
        fun buildIntent(activity: MainActivity?, id: String?): Intent {
            val intent = Intent(activity, CommentActivity::class.java)
            intent.putExtra("id", id)
            return intent
        }
    }
}